### PR TITLE
examples: Pin Rust nightly to 2023-01-10

### PR DESCRIPTION
**Pull request by vadorovsky**
We need to wait for the following fixes in Rust nightly to be able to use the newest one:

* https://github.com/rust-lang/rust/pull/106796
* https://github.com/rust-lang/rust/pull/106856

Signed-off-by: Michal Rostecki <vadorovsky@gmail.com>

### CODE REVIEW

File: examples/aya-tool/xtask/src/build_ebpf.rs [hunk 1]
```diff
@@ -52,8 +51,14 @@ pub fn build_ebpf(opts: Options) -> Result<(), anyhow::Error> {
...
         opts.profile.as_str(),
     ];
+
+    // Command::new creates a child process which inherits all env variables. This means env
+    // vars set by the cargo xtask command are also inherited. RUSTUP_TOOLCHAIN is removed
+    // so the rust-toolchain.toml file in the -ebpf folder is honored.
+
     let status = Command::new("cargo")
         .current_dir(&dir)
+        .env_remove("RUSTUP_TOOLCHAIN")
```
Comment (alessandrod) [inline examples/aya-tool/xtask/src/build_ebpf.rs:61]: I think you should do this in a separate PR, so when nightly is fixed we can revert this one and still keep this fix

Comment (vadorovsky) [reply examples/aya-tool/xtask/src/build_ebpf.rs:61]: What about separate commit?

Comment (vadorovsky) [reply examples/aya-tool/xtask/src/build_ebpf.rs:61]: I've pushed it as a separate commit. I don't really want to submit a new PR, because a standalone "pin nightly" PR would need to modify the same file to replace `+nightly` with `+nightly-2023-01-10`, quite a hassle IMO.

A separate commit is going to be easy enough to revert.
